octodns/provider/route53.py

```diff
@@ -957,11 +954,12 @@ def _process_desired_zone(self, desired):
                         rule.data['geos'] = filtered_geos
                     rules.append(rule)
 
-                dynamic.rules = rules
+                if rules != dynamic.rules:
+                    record = record.copy()
+                    record.dynamic.rules = rules
+                    desired.add_record(record, replace=True)
```

Comment:
It might nice to provide a helper for this. Something like:
```
desired.replace_record(record, rules=new_rules)
```
which will internally make a copy of the passed record with overridden attributes.

Or probably provide a more convenient `record.copy`:
```
record = record.copy(rules=new_rules)
```
Something to think about for another PR.

Reply:
I think in order to be generic and allow it to work with arbitrary records it'd look something like:

```python
record = record.copy(data={
    'dynamic': {
        'rules': new_rules,
    }
})
```

Basically the implementation (pseudo code) of it would look something like:

```
def copy(self, data):
    merged = self.data.deep_merging_update(data)
    return Record.new(self.zone, self.name, merged)
```

Reply:
Going to keep this as a todo. I think it'd be handy.